pkg/lib/authenticationflow/declarative/utils_common.go

```diff
@@ -436,6 +441,10 @@ func getAuthenticationOptionsForLogin(ctx context.Context, deps *authflow.Depend
 		}
 	}
 
+	if recoveryCodeBranch != nil && userHasRecoveryCode && (len(options) > 0) {
```

Comment:
The last condition `(len(options) > 0)` should be `len(secondaryAuthenticators) > 0`

Reply:
I think no, because `secondaryAuthenticators` includes authenticators which are disabled in config. `len(options)` is correct because at this point it is options excluding recovery code.